lib/auth/signup/sign_up_form.dart

```diff
@@ -41,73 +42,54 @@ class _SignUpFormState extends State<SignUpForm> {
         key: _formKey,
         child: Column(
           children: <Widget>[
-            TextFormField(
-              key: Key(SignUpKeys.usernameTextFromField),
-              controller: _usernameController,
-              validator: UsernameValidator.validate,
-              decoration: InputDecoration(
-                  labelText: SignUpStrings.usernameTextFormFieldText,
-                  labelStyle: TextStyle(
-                      color: Colors.grey
-                  )
-              ),
-            ),
-            TextFormField(
-              key: Key(SignUpKeys.emailTextFromField),
-              controller: _emailController,
-              validator: EmailValidator.validate,
-              decoration: InputDecoration(
-                  labelText: SignUpStrings.emailTextFormFieldText,
-                  labelStyle: TextStyle(
-                      color: Colors.grey
-                  )
-              ),
-            ),
-            TextFormField(
-              key: Key(SignUpKeys.passwordTextFromField),
-              controller: _passwordController,
-              validator: PasswordValidator.validate,
-              obscureText: true,
-              decoration: InputDecoration(
-                  labelText: 'Password',
-                  labelStyle: TextStyle(
-                      color: Colors.grey
-                  )
-              ),
-            ),
-            TextFormField(
-              key: Key(SignUpKeys.repeatedPasswordTextFromField),
-              validator: (value) {
-                if(_passwordController.text != value){
-                  return "Passwords doesn't match";
-                }
-                else{
-                  return null;
-                }
-              },
-              controller: _repeatedPasswordController,
-              obscureText: true,
-              decoration: InputDecoration(
-                  labelText: 'Repeat password',
-                  labelStyle: TextStyle(
-                      color: Colors.grey
-                  )
-              ),
-            ),
+            _createTextFormField(
+                SignUpKeys.usernameTextFromField,
```

Comment:
Zgodnie z dokumentacją: 

> When a [controller] is specified, [initialValue] must be null (the
> default). If [controller] is null, then a [TextEditingController]
> will be constructed automatically and its `text` will be initialized
> to [initialValue] or the empty string.

Nie ma potrzeby deklarowania kontrolerów ani podawania ich jako parametr. Zdefiniuj każdy widget jako pole klasy i podaj tutaj jako argumenty

Reply:
Wydaje mi się, że najlepiej będzie utworzyć osobne klasy. Będzie to jakoś poukładane i reużywalne chociaż